### PR TITLE
Compile datatable with MSVC

### LINKED ISSUE #1369: Make datatable compile with MSVC
This is a prerequisite for #1114 

Currently, there seems to be multiple compilation issues, including MSVC [not working properly with macros](https://blogs.msdn.microsoft.com/vcblog/2018/07/06/msvc-preprocessor-progress-towards-conformance/), and also lack of `sys/mman.h` library on windows (for memory-mapping)

### PR DESCRIPTION
**Pull request by oleksiyskononenko**
Datatable now compiles out of the box on Windows 10, provided that there is MSVC 2019 and WINSDK 10 installed. Some tests are failing or hanging though. These will be addressed in subsequent PRs.

Closes #1369

### CODE REVIEW

File: ci/xbuild/compiler.py [hunk 1]
```diff
@@ -240,6 +288,36 @@ def compile(self, src, obj, silent=False):
...
+        if not os.path.isdir(path):
+            raise ValueError("Lib directory %s not found" % path)
+        self._lib_dirs.append(path)
+        self._linker_flags += self._flags_for_lib_dir(path)
+        self.log.report_lib_dir(path)
+
+
+    def add_default_python_lib_dir(self):
+        if not self.is_msvc():
+            return
```
Comment (st-pasha) [inline ci/xbuild/compiler.py]: remove the check: already included under is_msvc() in ext.py

File: ci/xbuild/compiler.py [hunk 2]
```diff
@@ -240,6 +288,36 @@ def compile(self, src, obj, silent=False):
...
+
+        py_lib_dir = sysconfig.get_config_var("INCLUDEPY")
+        py_lib_dir += "\\..\\libs"
```
Comment (st-pasha) [inline ci/xbuild/compiler.py]:
```suggestion
        py_lib_dir = os.path.join(py_lib_dir,  "..", "libs")
```

**Comment by dmillar** [comment]
This is excellent! Any chance we could get packages with pyd distributed here?

**Comment by oleksiyskononenko** [comment]
@dmillar We still have some work to do in order to have a fully functional datatable build on Windows. It is in the road map for the next datatable release, here is a tracking issue #2281

**Pull request merged**